src/Equinox.MessageDb/MessageDb.fs

```diff
@@ -392,24 +387,20 @@ type private StoreCategory<'event, 'state, 'req>(context: MessageDbContext, code
                 let! token, state = context.Reload(log, streamName, requireLeader, pos, codec.Decode, fold, state, ct)
                 return struct(token, state)
             | ValueNone -> return! context.LoadBatched(log, streamName, requireLeader, codec.Decode, fold, initial, ct) }
-        | AccessStrategy.AdjacentProjection(_, access) -> loadAlgorithm access log category streamId streamName requireLeader ct
     let reload (log, sn, leader, token, state) ct = context.Reload(log, sn, leader, token, codec.Decode, fold, state, ct)
     interface ICategory<'event, 'state, 'req> with
         member _.Empty = context.TokenEmpty, initial
         member _.Load(log, categoryName, streamId, streamName, _maxAge, requireLeader, ct) =
-            loadAlgorithm access log categoryName streamId streamName requireLeader ct
+            loadAlgorithm log categoryName streamId streamName requireLeader ct
         member x.Sync(log, categoryName, streamId, streamName, req, token, state, events, ct) = task {
             let encode e = codec.Encode(req, e)
             let encodedEvents: IEventData<EventBody>[] = events |> Array.map encode
             let state' = fold state events
-            let project =
-                match access with
-                | AccessStrategy.AdjacentProjection(project, _) -> Some (project (StreamId.Elements.trust streamId) state')
-                | _ -> None
-            match! context.TrySync(log, categoryName, streamId, streamName, token, encodedEvents, project, ct) with
+            let onSync = match onSync with Some f -> Some (fun conn -> f (StreamId.Elements.trust streamId) state' conn) | None -> None
```

Comment:
I love one liners too but maybe put the None first
and you promised to supply the version, no?